core/trino-main/src/main/java/io/trino/metadata/MetadataListing.java

```diff
@@ -132,22 +134,54 @@ private MetadataListing() {}
 
     public static Map<SchemaTableName, List<ColumnMetadata>> listTableColumns(Session session, Metadata metadata, AccessControl accessControl, QualifiedTablePrefix prefix)
     {
-        Map<SchemaTableName, List<ColumnMetadata>> tableColumns = metadata.listTableColumns(session, prefix).entrySet().stream()
-                .collect(toImmutableMap(entry -> entry.getKey().asSchemaTableName(), Entry::getValue));
+        Map<SchemaTableName, Optional<List<ColumnMetadata>>> tableColumns = metadata.listTableColumns(session, prefix).values().stream()
```

Comment:
Why `Optional<List<ColumnMetadata>>`? If table has no columns, there we should not list it at all as part of `tableColumns`

Reply:
The connector needs to tell the engine that the table is redirected so the engine needs to find the columns from the redirected table. e.g. If user lists columns for table `A` that's redirected to `B`, they should get the metadata from B, but the table name displayed is still `A`.